client/src/RedeemPage/index.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { FC, useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+import delve from 'dlv';
+import { useToasts } from 'react-toast-notifications';
+import classNames from 'classnames';
+import { Formik, Form, Field, FieldProps } from 'formik';
+
+// components
+import { ScanHeader, ScanFooter } from '../ScanPage';
+import { SubmitButton } from '../components/SubmitButton';
+
+// lib
+import { RedeemSchema } from '../lib/schemas';
+
+// api
+import { redeemWithEmail, TokenInfo, getTokensFor } from '../api';
+import { RouteComponentProps } from 'react-router-dom';
+
+type RedeemFormValues = {
+  address: string;
+};
+
+const initialValues = { address: '' };
+
+export const RedeemPage: FC<RouteComponentProps> = ({ match }) => {
+  // react hooks
+  const [isRedeemLoading, setIsRedeemLoading] = useState<boolean>(false);
+  const [tokens, setTokens] = useState<TokenInfo[]>([]);
+
+  // lib hooks
+  const { addToast } = useToasts();
+
+  // constants
+  const uid = delve(match, 'params.uid');
+  console.log('uid ', uid);
+  const email = delve(match, 'params.email');
+  console.log('email', email);
+
+  // effects
+  useEffect(() => {
+    getTokens();
+  }, []) // eslint-disable-line
+
+  // methods
+  const getTokens = async () => {
+    try {
+      // TODO: Uncomment next line when backend supports email
+      // How do we get the email? Will `/actions/scan/${email}` be a valid endpoint?
+      // const tokens = await getTokensFor(email);
+
+      // TODO: Remove next line when backend supports email
+      const mockAddress = '0x5A384227B65FA093DEC03Ec34e111Db80A040615';
+      const tokens = await getTokensFor(mockAddress);
+
+      setTokens(tokens);
+    } catch (error) {
+      addToast(error.message, {
+        appearance: 'error',
+        autoDismiss: false,
+      });
+    }
+  };
+
+  // handlers
+  const handleForm = async (values: RedeemFormValues) => {
+    setIsRedeemLoading(true);
+    const { address } = values;
+
+    redeemWithEmail(address, uid)
+      .then(() => {
+        setIsRedeemLoading(false);
+
+        addToast('POAPs loaded successfully', {
+          appearance: 'error',
+          autoDismiss: false,
+        });
+      })
+      .catch((error) => {
+        addToast(error.message, {
+          appearance: 'error',
+          autoDismiss: false,
+        });
+      });
```

Comment:
Shouldn't we move `setIsRedeemLoading(false);` to finally since if an error has been thrown the spinner will be indefinitely being shown

```suggestion
    redeemWithEmail(address, uid)
      .then(() => addToast('POAPs loaded successfully', { appearance: 'error', autoDismiss: false }))
      .catch((error) => addToast(error.message, { appearance: 'error', autoDismiss: false }))
      .finally(() => setIsRedeemLoading(false));
```

Reply:
Totally. Thanks